chutney/ui/src/app/shared/components/execute-modal/scenario-execute-modal.component.ts

```diff
@@ -39,27 +50,82 @@ export class ScenarioExecuteModalComponent implements OnInit {
     constructor(
         private datasetService: DataSetService,
         private environmentService: EnvironmentService,
-        private eventManagerService: EventManagerService,
-        private translateService: TranslateService) {
+        private formBuilder: FormBuilder,
+        private translateService: TranslateService,
+        private changeDetectorRef: ChangeDetectorRef){
     }
 
 
     ngOnInit(): void {
-        this.datasetService.findAll().subscribe((res: Array<Dataset>) => {
-            this.datasets = res;
-            this.filteredDatasets = res;
+        this.translateService.get('dataset.actions.createDataset').pipe(
+            map((res: string) => this.createDataset = new Dataset(res, "", [], new Date(), [], [])),
+            map(createDataset => {
+                this.datasetService.findAll().subscribe((res: Array<Dataset>) => {
+                    this.datasets = [...res, createDataset];
+                    this.filteredDatasets = [...res, createDataset];
+            });
+        })).subscribe();
+
+        this.datasetForm = this.formBuilder.group({
+            saveDatasetName: new FormControl({ value: '', disabled: true }),
+            saveDatasetCheckbox: [false, Validators.required],
+            keyValues: new FormControl(),
+            multiKeyValues: new FormControl()
         });
+
         this.environmentService.names().subscribe((res: string[]) => {
             this.environments = res;
             if (this.environments.length === 1) {
                 this.selectedEnv = this.environments[0];
             }
+            this.environmentsLoaded = true;
         });
+
+        this.saveDatasetCheckboxChanged();
+    }
+
+    selectedDatasetChanged() {
+        if (this.selectedDataset == this.createDataset) {
+            this.editDataset(null, this.selectedDataset, true);
+        } else {
+            this.editionDataset = false;
+            this.changeModalSize("lg")
+        }
     }
 
-    execute() {
+    executeModal() {
+        if (this.editionDataset) {
+            const editedDataset = this.buildDataset();
+            if (this.validateDatasetCreation(editedDataset)) {
+                this.translateService.get("scenarios.execution.modal.error.invalidDataset").subscribe(res => {
```

Comment:
- use `translateService.instant` instead to have more readable code
- call this.execute(this.selectedDataset) **once**. prepare dataset (and save it if needed ) before